tests/system/src/test/java/test/robot/javafx/scene/RobotTest.java

```diff
@@ -167,23 +372,124 @@ public void testMouseClick() {
+        if (mouseAction == MouseAction.PRESSED) {
+            Util.runAndWait(() -> robot.mouseRelease(MouseButton.PRIMARY, MouseButton.SECONDARY, MouseButton.MIDDLE));
+        }
+    }
+
+    @Test
+    public void testMousePressThrowsISEOnWrongThread() {
+        try {
             robot.mousePress(MouseButton.PRIMARY);
+        } catch (IllegalStateException e) {
+            return;
+        }
+        Assert.fail("Expected IllegalStateException");
+    }
+
+    @Test
+    public void testMousePressThrowsNPEForNullArgument() {
+        Util.runAndWait(() -> {
+            try {
+                robot.mousePress(null);
+            } catch (NullPointerException e) {
+                return;
+            }
+            Assert.fail("Expected NullPointerException");
+        });
+    }
+
+    @Test
+    public void testMouseReleaseThrowsISEOnWrongThread() {
+        try {
             robot.mouseRelease(MouseButton.PRIMARY);
+        } catch (IllegalStateException e) {
+            return;
+        }
+        Assert.fail("Expected IllegalStateException");
+    }
+
+    @Test
+    public void testMouseReleaseThrowsNPEForNullArgument() {
+        Util.runAndWait(() -> {
+            try {
+                robot.mouseRelease(null);
+            } catch (NullPointerException e) {
+                return;
+            }
+            Assert.fail("Expected NullPointerException");
+        });
+    }
+
+    @Test
+    public void testMouseClickThrowsISEOnWrongThread() {
+        try {
+            robot.mouseClick(MouseButton.PRIMARY);
+        } catch (IllegalStateException e) {
+            return;
+        }
+        Assert.fail("Expected IllegalStateException");
+    }
+
+    @Test
+    public void testMouseClickThrowsNPEForNullArgument() {
+        Util.runAndWait(() -> {
+            try {
+                robot.mouseClick(null);
+            } catch (NullPointerException e) {
+                return;
+            }
+            Assert.fail("Expected NullPointerException");
         });
-        waitForLatch(onClickLatch, 5, "Timeout while waiting for button.onMouseClicked().");
-        Assert.assertEquals("primary mouse button should be clicked by Robot", "Clicked", button.getText());
     }
 
     @Test
-    public void testMouseWheel() {
+    public void testMouseWheelPositiveAmount() {
+        testMouseWheel(5);
+    }
+
+    @Test
+    public void testMouseWheelNegativeAmount() {
+        testMouseWheel(-5);
+    }
+
+    private static void testMouseWheel(int amount) {
         CountDownLatch onScrollLatch = new CountDownLatch(1);
         CountDownLatch setSceneLatch = new CountDownLatch(1);
         Button button = new Button("Scroll me");
         InvalidationListener invalidationListener = observable -> setSceneLatch.countDown();
+        int[] totalScroll = new int[]{0};
+        long[] firstScrollMillis = new long[]{0};
+        int scrollMultiplier = System.getProperty("os.name").toLowerCase().startsWith("mac") ? -1 : 40;
```

Comment:
Can you add a comment here? This suggests that there might be an issue with scrolling in the Mac robot implementation. We might want to file a follow-up issue to investigate.